pkg/xds/bootstrap/template_v3.go

```diff
@@ -46,6 +46,30 @@ func genConfig(parameters configParameters, useTokenPath bool) (*envoy_bootstrap
 		features = append(features, feature)
 	}
 
+	runtimeLayers := []*envoy_bootstrap_v3.RuntimeLayer{{
+		Name: "kuma",
+		LayerSpecifier: &envoy_bootstrap_v3.RuntimeLayer_StaticLayer{
+			StaticLayer: util_proto.MustStruct(map[string]interface{}{
+				"envoy.restart_features.use_apple_api_for_dns_lookups": false,
+				"re2.max_program_size.error_level":                     4294967295,
+				"re2.max_program_size.warn_level":                      1000,
+			}),
+		},
+	}}
+
+	if parameters.IsGatewayDataplane {
+		runtimeLayers = append(runtimeLayers,
+			&envoy_bootstrap_v3.RuntimeLayer{
+				Name: "gateway",
+				LayerSpecifier: &envoy_bootstrap_v3.RuntimeLayer_StaticLayer{
+					StaticLayer: util_proto.MustStruct(map[string]interface{}{
+						"overload.global_downstream_max_connections": 50000,
```

Comment:
I think we can make it configurable, and just leave some default.

Reply:
I agree. Making it configurable would be easy enough.

Reply:
Control plane wide or per proxy?

Reply:
Good question. I'm guessing per proxy as you may want to have different gateways with different values. Except if you'd use a different policy to do this kind of connection limits and in which case that's a CP config. WDYT?

Reply:
Not sure. Definitely per proxy makes sense as a feature but feels like it might be part of a larger story for all kinds of gateway-specific performance customization.

Reply:
Yes it's part of the IncomingRequestTimeout policy we've talked about in the past. So let's just make that a CP config

Reply:
Added @lahabana @lukidzi